templates/_helpers.tpl

```diff
@@ -58,6 +58,32 @@ Compute if the server is enabled.
   (and (eq (.Values.server.enabled | toString) "-") (eq (.Values.global.enabled | toString) "true"))) -}}
 {{- end -}}
 
+{{/*
+Compute if the server auth delegator serviceaccount is enabled.
+*/}}
+{{- define "vault.serverServiceAccountEnabled" -}}
+{{- $_ := set . "serverServiceAccountEnabled"
+  (and
+    (eq (.Values.server.serviceAccount.create | toString) "true" )
+    (or
+      (eq (.Values.server.enabled | toString) "true")
```

Comment:
Trying to figure out how `server.enabled`, `global.enabled`, `serviceAccount.create`, and `authDelegator.enabled` all **should** interact is a good way to get a headache.

This is what I came up with.